### DISCUSSION TITLE
PnP Modern Search Web Part showing results on page load or when search box is empty

### DISCUSSION THREAD

**Discussion by MrWorkingCat** [microsoft-search/pnp-modern-search#1882] [Q&A]
I am new to the PnP Modern Search web part, but I see it may be very useful improving the UX of modern SharePoint sites in out organisation. 

I am using latest release of PnP Modern Search as per March 2022 (v4.5.4)

Installed one week ago in a O365 Developer environment.

After installation, I followed steps in [Scenario 1: Create a simple search page in the documentation](https://microsoft-search.github.io/pnp-modern-search/scenarios/create-simple-search-page/).

At the end of the tutorial, I ended up with a search that behaves differently as per the tutorial. Specifically, the issue is that on page load, or when search box is empty, the search results web part returns all results, similar to "*".

Is anybody else getting this issue or knows how to fix it?

Note: Similar questions has been asked outside this site, but none of the replies in those discussions worked for me.

- [SharePoint Search prevent showing results when loading page](https://sharepoint.stackexchange.com/questions/95541/sharepoint-search-prevent-showing-results-when-loading-page) 

- [Search Results web part populating without any search terms being entered into Search Box web part](https://sharepoint.stackexchange.com/questions/201713/search-results-web-part-populating-without-any-search-terms-being-entered-into-s)

**Comment by wobba** [comment]
If the query template shows `{searchTerms}` and you haven't checked **Use a default value** in the connection setup, you should not see results.

**Comment by MrWorkingCat** [comment]
Thanks. I am getting results on page load or when search box is empty with the configuration below:

![image](https://user-images.githubusercontent.com/4206482/156674577-6e91794b-f859-4d82-9133-3e27454111d7.png)

![image](https://user-images.githubusercontent.com/4206482/156674630-01f567db-f073-4b54-bf5f-c82402419c51.png)

The only way I can avoid those results is with a workaround like the one below, which I always thought and you seem to confirm, is not the right approach:

![image](https://user-images.githubusercontent.com/4206482/156674977-6a8745d7-f0c5-4d15-a3c0-9c76ebf7dd15.png)

**Reply by wobba** [reply]
Hi, I see the issue now for some tenants, and I believe I've seen internal Microsoft information around this. I'll see if I can track it down.
